Backend

MaxSukhovetskyi opened the pull request:
lol kek cheburek

MaxSukhovetskyi committed:
Separated api routes

MaxSukhovetskyi committed:
Fixes to previous commit

MaxSukhovetskyi committed:
Refactored cities api stracture

Commit:
Merged dev into backend

MaxSukhovetskyi committed:
Removed old cities_controler

rapkin commented on README.md:
```diff
@@ -9,14 +9,15 @@
 
 * Install Node.js
-* `npm install -g bower gulp coffee-script`
+* `npm install -g bower gulp coffee-script pug stylus`
```
look, у нас для цих пакетів є package.json. поняв this?

rapkin commented on helpers/database.py:
```diff
@@ -6,23 +6,23 @@
+    user="postgres",
     host="127.0.0.1",
-    password='data-vis',
+    password='root',
```
а пользоватєля і password настрой (добав) в себе (а ідеально в конфіг це винести - config.json який потім в .gitignore добавити)

rapkin reviewed (commented):
Карочє все ок, тільки перемісти назад файли sql і export (хай будуть в корні - зачєм ти їх в хелпери переніс?)

MaxSukhovetskyi commented:
Бо вони тільки корень засоряють. Вони рідко вкористовуються да і логічно повинні бути розміщені в хелперах.

rapkin commented:
@Grindl4242 їх логінчіше (якщо вже виносити кудись) винести в папку data. в хелперах тільки код має бути

MaxSukhovetskyi committed:
Refactored config functionality

MaxSukhovetskyi committed:
Add port in global config

MaxSukhovetskyi committed:
Add db connection to app_context, refactored db api

MaxSukhovetskyi committed:
Add dictionary sql query results

MaxSukhovetskyi committed:
Fixed config load

rapkin commented:
@Grindl4242  а чо конфіги в пітоні (ми ж домовились шо в json)? і зроби всі порти циферками.
конфіг хай парсить і оголошує глобально, всі поля можеш зробити великими літерами (якщо в пітоні так прийнято)

MaxSukhovetskyi committed:
Refactored db connection

Bind connection config to global config, add db conn to global scope, refactored metod of init database.

MaxSukhovetskyi commented:
Так можна юзать наслідування, загрузку конфіга в 1 строку і не треба писать додаткову логіку для обробки json. Да й нащо тобі в корені сервера конфіг в json?

rapkin commented:
@Grindl4242 ну я ж кажу що конфіг хай оголошує глобально (можеш ще в хелпери винести роботу з конфігом), мені прост json треба буде потім

MaxSukhovetskyi committed:
Revert config api

MaxSukhovetskyi committed:
Merge branch 'database_refact' into backend

MaxSukhovetskyi committed:
Revert config functionality back to json

MaxSukhovetskyi committed:
Polished config api

MaxSukhovetskyi committed:
Reworked folder structure, prepare place for auth subapp

MaxSukhovetskyi committed:
Polished root finder for helpers, applied api.md

Commit:
Merge branch 'new_folder_struct' into backend

rapkin commented:
нащо ти закрив, його треба доробити і вмерджити
